### PR TITLE
Create new <Time> component, remove date-fns from most other components

### PR DESCRIPTION
**Pull request by mattaho**
The intent of this is to be able to pass in anything as the `pubDate` prop for components that need one, so that the pubdate can be formatted however we need. In addition to this I created a `<Time>` component to handle formatting dates/times more easily.

This means that when using, for example, a Teaser component, we should pass in the `<Time>` component as the pubDate prop, like:

```
<Teaser pubDate={<Time dateTime={story.pubdate} />} />
```

### CODE REVIEW

File: src/molecules/Teaser/Teaser.js
Comment (kimthompson) [inline src/molecules/Teaser/Teaser.js]: I can't remember seeing or needing a "check if I exist and, if so, show me" command in other React projects. Is this necessary?

**Review by kimthompson — APPROVED** [review]
I found one possible redundancy, but the rest of this looks great. Feel free to merge this in and publish when you're ready! I'll hold off on my Amat junk until I write tests for it tomorrow morning.

File: src/molecules/Teaser/Teaser.js
Comment (mattaho) [reply src/molecules/Teaser/Teaser.js]: You're right, I don't need the check for this; react just won't show it if it doesn't exist. Since there's no html inside this anymore we don't need the check. I'll change that and merge.

**Pull request merged**
